src/etl/gwas_ingest/study_ingestion.py

```diff
@@ -294,30 +486,84 @@ def ingest_gwas_catalog_studies(
     # Read GWAS Catalogue raw data
     gwas_studies = read_study_table(etl, study_file)
     gwas_ancestries = parse_ancestries(etl, ancestry_file)
+    ss_studies = get_sumstats_location(etl, summary_stats_list)
 
+    # Sample size extraction is a separate process:
     study_size_df = extract_discovery_sample_sizes(gwas_studies)
-    ss_studies = get_sumstats_location(etl, summary_stats_list)
 
-    studies = (
+    # Mapping ancestries of the discover sampe description to GnomAD:
+    # gnomad_ancestries = parse_gnomad_ancestries(
+    #     gwas_ancestries.select("studyAccession", "discoverySamples").distinct()
+    # )
+
+    return (
         gwas_studies
         # Add study sizes:
         .join(study_size_df, on="studyAccession", how="left")
+        # Adding GnomAD ancestry mapping:
+        # .join(gnomad_ancestries, on="studyAccession", how="left")
         # Adding summary stats location:
-        .join(
-            ss_studies,
-            on="studyAccession",
-            how="left",
-        )
+        .join(ss_studies, on="studyAccession", how="left")
         .withColumn("hasSumstats", f.coalesce(f.col("hasSumstats"), f.lit(False)))
         .join(gwas_ancestries, on="studyAccession", how="left")
+        # Select relevant columns:
         .select(
-            "*",
-            parse_efos("mappedTraitUri").alias("efos"),
+            "studyAccession",
+            # Publication related fields:
+            "pubmedId",
+            "firstAuthor",
+            "publicationDate",
+            "journal",
+            "study",
+            # Disease related fields:
+            "studyDiseaseTrait",
+            parse_efos("mappedTraitUri").alias("studyEfos"),
             parse_efos("mappedBackgroundTraitUri").alias("backgroundEfos"),
+            # Sample related fields:
+            "initialSampleSize",
+            "nCases",
+            "nControls",
+            "nSamples",
+            # Ancestry related labels:
+            "discoverySamples",
+            "replicationSamples",
+            # "gnomadSamples",
+            # Summary stats fields:
+            "summarystatsLocation",
+            "hasSumstats",
         )
-        .drop("initialSampleSize", "mappedBackgroundTraitUri", "mappedTraitUri")
+        .persist()
     )
 
-    validate_df_schema(studies, "studies.json")
 
-    return studies
+def generate_study_table(association_study: DataFrame) -> DataFrame:
```

Comment:
I did a revision of the study index schema ([here](https://gist.github.com/ireneisdoomed/da6e7ba344a50651ebbe860224907243#file-studies-py-L57)).
Can you please take a look and see if there are things that we can change? I'd like to have them harmonised.

Reply:
OK, I'll take a look and update.